pybaseball/statcast_pitcher_spin.py

```diff
@@ -1,140 +1,200 @@
+        'ay', 'az', 'release_spin_rate']].copy()
+
+    spin_df = find_intermediate_values(spin_df)
+
+    pitcher_data[['Mx', 'Mz', 'phi', 'theta']] = spin_df[[
+        'Mx', 'Mz', 'phi', 'theta']].copy()
+
+    return pitcher_data
 
 # def get_statcast_pither_test_data():
 # 	df = pd.read_csv("tests/statcast_pitching_test_data.csv")
 # 	return df
 
+
+def find_intermediate_values(spin_df):
+    """Calls each intermediate function in sequence"""
+    spin_df = find_release_point(spin_df)
+    spin_df = find_release_time(spin_df)
+    spin_df = find_release_velocity_components(spin_df)
+    spin_df = find_flight_time(spin_df)
+    spin_df = find_average_velocity_components(spin_df)
+    spin_df = find_average_velocity(spin_df)
+    spin_df = find_average_drag(spin_df)
+    spin_df = find_magnus_acceleration_magnitude(spin_df)
+    spin_df = find_average_magnus_acceleration(spin_df)
+    spin_df = find_magnus_magnitude(spin_df)
+    spin_df = find_phi(spin_df)
+    spin_df = find_lift_coefficient(spin_df)
+    spin_df = find_spin_factor(spin_df)
+    spin_df = find_transverse_spin(spin_df)
+    spin_df = find_spin_efficiency(spin_df)
+    spin_df = find_theta(spin_df)
+
+    return spin_df
+
+
 def find_release_point(df):
-	df['yR'] = (60.5 - df['release_extension']).round(SIG_DIG)
-	return df
+    df['yR'] = (DISTANCE_FROM_HOME_TO_MOUND - df['release_extension']).round(SIG_DIG)
+    return df
+
 
 def find_release_time(df):
-	df['tR'] = time_duration(df['yR'], df['vy0'], df['ay'], 50, False).round(SIG_DIG)
-	return df
+    df['tR'] = time_duration(
+        df['yR'],
+        df['vy0'],
+        df['ay'],
+        DISTANCE_TO_PLATE_AT_VELOCITY_CAPTURE,
+        False).round(SIG_DIG)
+    return df
+
 
 def find_release_velocity_components(df):
-	df['vxR'] = (df['vx0'] + (df['ax'] * df['tR'])).round(SIG_DIG)
-	df['vyR'] = (df['vy0'] + (df['ay'] * df['tR'])).round(SIG_DIG)
-	df['vzR'] = (df['vz0'] + (df['az'] * df['tR'])).round(SIG_DIG)
-	return df
+    df['vxR'] = (df['vx0'] + (df['ax'] * df['tR'])).round(SIG_DIG)
+    df['vyR'] = (df['vy0'] + (df['ay'] * df['tR'])).round(SIG_DIG)
+    df['vzR'] = (df['vz0'] + (df['az'] * df['tR'])).round(SIG_DIG)
+    return df
+
 
 def find_flight_time(df):
-	df['tf'] = time_duration(df['yR'], df['vyR'], df['ay'], 17/12, True).round(SIG_DIG)
-	return df
+    df['tf'] = time_duration(
+        df['yR'],
+        df['vyR'],
+        df['ay'],
+        Y_VALUE_AT_FINAL_MEASUREMENT,
+        True).round(SIG_DIG)
+    return df
+
 
 def find_average_velocity_components(df):
-	df['vxbar'] = (2*df['vxR'] + df['ax']*df['tf'])/2
-	df['vybar'] = (2*df['vyR'] + df['ay']*df['tf'])/2
-	df['vzbar'] = (2*df['vzR'] + df['az']*df['tf'])/2
+    df['vxbar'] = (2*df['vxR'] + df['ax']*df['tf'])/2
+    df['vybar'] = (2*df['vyR'] + df['ay']*df['tf'])/2
+    df['vzbar'] = (2*df['vzR'] + df['az']*df['tf'])/2
+
+    df['vxbar'] = df['vxbar'].round(4)
+    df['vybar'] = df['vybar'].round(4)
+    df['vzbar'] = df['vzbar'].round(4)
```

Comment:
I'm not sure that we want to be rounding, but if we are, this should be `SIG_DIG`